File: kakao/level1/크레인_인형뽑기_게임.py
def solution(board, moves):
    answer = 0
    tmp = []
    L = len(board)
    for m in moves:
        for i in range(L):
            if board[i][m-1] != 0:
                tmp.append(board[i][m-1])
                board[i][m-1] = 0
                break
        
        if len(tmp) > 1 and tmp[-1] == tmp[-2]:
            tmp = tmp[:-2]
            answer += 2

    return answer


board = [[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]]
moves = [1,5,3,5,1,2,1,4]
print(solution(board, moves)) # 4

File: Java/programmers/kakao/level1/크레인인형뽑기게임.java
package CodeTest.Java.programmers.kakao.level1;

import java.util.*;

public class 크레인인형뽑기게임 {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> tmp = new Stack<>();
        for (int m : moves) {
            for (int i=0; i<board.length; i++) {
                if (board[i][m-1] != 0) {
                    if (!tmp.isEmpty() && board[i][m-1] == tmp.peek()) {
                        answer += 2;
                        tmp.pop();
                        board[i][m-1] = 0;
                        break;
                    }
                    tmp.push(board[i][m-1]);
                    board[i][m-1] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}
